### PR TITLE
Portal connection

### PR DESCRIPTION
**Pull request by stevenernst131**
This change allows for blades to be opened in the portal (state we were in on friday).

Also added support for application insights blade

### CODE REVIEW

File: app/shared/models/portal.ts [hunk 1]
```diff
@@ -107,27 +107,27 @@ export class SupportBladeDefinition {
 
 export class SupportBladeDefinitions {
     public static Pulse: SupportBladeDefinition = {
-        Identifier: BladeOptionsEnum.pulse.toString(),
+        Identifier: BladeOptionsEnum[BladeOptionsEnum.pulse],
```
Comment (stevenernst131) [inline app/shared/models/portal.ts]: I acknowledge that this is a bit awkward, plan to change it.

File: app/availability/analysis/app-analysis.component.ts [hunk 2]
```diff
@@ -28,7 +29,7 @@ export class AppAnalysisComponent implements OnInit {
     metSLA: boolean;
     abnormalTimePeriod: IAbnormalTimePeriod;
 
-    constructor(private _route: ActivatedRoute, private _appAnalysisService: AppAnalysisService) {
+    constructor(private _route: ActivatedRoute, private _appAnalysisService: AppAnalysisService, private _portalActionService: PortalActionService) {
```
Comment (ShekharGupta1988) [inline app/availability/analysis/app-analysis.component.ts]: I dont see this service getting used anywhere right now in this component.

File: app/shared/models/solution-ui-model/openbladesolution.ts [hunk 3]
```diff
@@ -2,27 +2,28 @@ import { SolutionUIModelBase } from './solution-ui-model-base';
 import { SolutionProperties, SubAction, SolutionMetadata } from './solutionproperties';
 import { SolutionType, ActionType, ActionStatus } from '../enumerations';
 import { INameValuePair } from '../namevaluepair';
+import { SupportBladeDefinitions } from '../portal';
+import { PortalActionService } from '../../services';
 
 export abstract class OpenBladeSolutionUIModel extends SolutionUIModelBase {
 
     protected abstract _bladeUrl: string;
```
Comment (ShekharGupta1988) [inline app/shared/models/solution-ui-model/openbladesolution.ts]: You can remove the _bladeUrl. I just added it for testing. Was not sure how the blade was opened earlier.

File: app/availability/analysis/app-analysis.component.ts (hunk 2 above)
Comment (stevenernst131) [inline app/availability/analysis/app-analysis.component.ts]: yup, leftover from previous testing

**Pull request merged**